base/env.jl

```diff
@@ -98,6 +98,51 @@ Base.EnvDict with "N" entries:
 """
 const ENV = EnvDict()
 
+"""
+    Base.get_bool_env(name::String, default::Bool = false)::Bool
+
+Evaluate whether the value of environnment variable `name` is truthy or falsy,
+and error if none are recognized. If the variable is not set, or is set to "",
+return `default`.
+
+Recognized values are:
+    truthy: "t", "true", "y", "yes", "1"
+    falsy:  "f", "false", "n", "no", "0"
+"""
+function get_bool_env(name::String, default::Bool = false)
+    haskey(ENV, name) || return default
+    val = lowercase(get(ENV, name, ""))
+    if isempty(val)
+        return default
+    elseif is_truthy(val)
+        return true
+    elseif is_falsy(val)
+        return false
+    else
+        error("""
+        Failed to parse the value `$val` of environment variable `$name` as Boolean.
+        Recognized values are:
+            truthy: "t", "true", "y", "yes", "1"
+            falsy:  "f", "false", "n", "no", "0"
+        """)
+    end
+end
+"""
+    Base.is_truthy(str::String)
+
+Evaluate whether a string is "truthy" i.e. representative of a `true`.
+Accepted values are ("t", "true", "y", "yes", "1")
+"""
+is_truthy(str::String) = str in ("t", "true", "y", "yes", "1")
```

Comment:
can we just move this line into `get_bool_env` and avoid having this as a separate function. Seems like it runs the risk of being mistakenly confused with the notion of "truthy" and "falsy" values (like "empty strings are falsy and non-empty are truthy) https://discourse.julialang.org/t/on-the-arbitrariness-of-truth-iness/81381

Reply:
huh yeah. Easiest to side-step that I think